public/components/metrics/redux/slices/metrics_slice.ts

```diff
@@ -123,56 +150,51 @@ export const metricSlice = createSlice({
       updateLayoutByDeSelection(state, payload);
       state.selected = state.selected.filter((id) => id !== payload.id);
     },
-    searchMetric: (state, { payload }) => {
-      state.searched = state.metrics.filter(
-        (metric: any) => metric.name.includes(payload.id) && !state.selected.includes(payload.id)
-      );
-    },
-    clearSearchedMetrics: (state, { payload }) => {
-      state.searched = [];
+    setSearch: (state, { payload }) => {
+      state.search = payload;
     },
+
     updateMetricsLayout: (state, { payload }) => {
       state.metricsLayout = payload;
+      state.selected = sortBy(payload, ['x', 'y']).map(({ id }) => id);
+    },
+    setDataSources: (state, { payload }) => {
+      state.dataSources = [OBSERVABILITY_CUSTOM_METRIC, ...payload];
+    },
+    setDataSourceTitles: (state, { payload }) => {
+      state.dataSourceTitles = ['Observability Custom Metrics', ...payload];
+    },
+    setDataSourceIcons: (state, { payload }) => {
+      state.dataSourceIcons = payload;
     },
-  },
-  extraReducers: (builder) => {
-    builder.addCase(loadMetrics.fulfilled, (state, { payload }) => {
-      state.metrics = payload;
-      state.searched = [];
-      filterDeletedLayoutIds(state, payload);
-    });
   },
 });
 
 export const {
+  setMetrics,
   deSelectMetric,
   selectMetric,
   updateMetricsLayout,
-  searchMetric,
-  clearSearchedMetrics,
+  setSearch,
+  setDataSources,
+  setDataSourceTitles,
+  setDataSourceIcons,
 } = metricSlice.actions;
 
-export const metricsStateSelector = (state) => state.metrics;
-
 export const availableMetricsSelector = (state) =>
-  state.metrics.metrics.filter(
-    (metric) => !state.metrics.selected.includes(metric.id) && !metric.recentlyCreated
-  );
+  state.metrics.metrics
+    .filter((metric) => !state.metrics.selected.includes(metric.id))
+    .filter((metric) => state.metrics.search === '' || metric.name.includes(state.metrics.search));
```

Comment:
https://github.com/opensearch-project/dashboards-observability/blob/108189754cb1b98c75cc2a7e6909b990f32687b6/public/components/metrics/sidebar/search_bar.tsx#L16

this only triggers when user is typing

If user didn't type in search bar, `state.metrics.search` is undefined, and this filter statement filter out all metrics?

Reply:
Updated initialState.search at top of file (ln: 42).   Default value empty string.
This change got lost in a merge.